### PR TITLE
fix(toml)!: Disallow `[lints]` in virtual workspaces

### PR DESCRIPTION
**Pull request by epage**
This was missed with the initial `[lints]` implementation.

While this is a breaking change, this is aligned with ones we've done in the past.  A lot of times, we warn first.  My hope is that isn't needed this time because
- It only exists virtual workspaces so they aren't published
- It is a nop to have this which is likely to be caught
- This is so new that the number of people using it, and likely running into this case, is quite low.

### CODE REVIEW

File: src/cargo/util_schemas/manifest.rs [hunk 1]
```diff
@@ -45,6 +45,7 @@ pub struct TomlManifest {
     pub workspace: Option<TomlWorkspace>,
     pub badges: Option<InheritableBtreeMap>,
     pub lints: Option<InheritableLints>,
+    // when adding new fields, be sure to check whether `to_virtual_manifest` should disallow them
```
Comment (weihanglo) [inline src/cargo/util_schemas/manifest.rs:48]: Wonder we have a way to catch this automatically.

**Review by weihanglo — APPROVED** [review]
Agree that we make this a hard error asap. The old PR did it as well <https://github.com/rust-lang/cargo/pull/6276>.

**Comment by weihanglo** [comment]
What does `!` mean in `fix(toml)!:`?

**Comment by epage** [comment]
> What does `!` mean in `fix(toml)!:`?

Breaking change

**Comment by jdahlstrom** [comment]
I'm probably missing something, but what's the rationale for this? Surely it may be useful to also harmonize lints in a virtual workspace?

**Comment by weihanglo** [comment]
> I'm probably missing something, but what's the rationale for this? Surely it may be useful to also harmonize lints in a virtual workspace?

Then one should use `[workspace.lints]` instead of `[lints]`. This PR prohibits the latter.

**Comment by GeeWee** [comment]
Ah, that distinction also confused me. I think it might be worth emphasizing it in the release notes if they're not immutable :)

**Comment by epage** [comment]
I posted #13425

**Pull request merged**